nym-wallet/src/pages/balance/index.tsx

```diff
@@ -9,9 +9,26 @@ import { TransferModal } from './components/TransferModal';
 
 export const Balance = () => {
   const [showTransferModal, setShowTransferModal] = useState(false);
+  const [showVestingCard, setShowVestingCard] = useState(false);
 
   const { userBalance } = useContext(AppContext);
 
+  useEffect(() => {
+    const { originalVesting, currentVestingPeriod, tokenAllocation } = userBalance;
+    if (originalVesting) {
+      setShowVestingCard(true);
+    }
+    if (
+      originalVesting &&
+      currentVestingPeriod === 'After' &&
+      tokenAllocation?.locked === '0' &&
+      tokenAllocation?.vesting === '0' &&
+      tokenAllocation?.spendable === '0'
+    ) {
+      setShowVestingCard(false);
+    }
```

Comment:
What about 

```
if (
      originalVesting &&
      currentVestingPeriod === 'After' &&
      tokenAllocation?.locked === '0' &&
      tokenAllocation?.vesting === '0' &&
      tokenAllocation?.spendable === '0'
    ) {
      setShowVestingCard(false);
    } else if (originalVesting) {
       setShowVestingCard(true)
    }
```
What you have done is fine and works but I think we should set it up so that it is either setShowVestingCard(true) or set showVestingCard(false). At the moment both of your if statements can be triggered if `originalVesting` is true